src/index.ts

```diff
@@ -110,4 +111,35 @@ export const dataProvider = (client: SanityClient): DataProvider => {
     },
   };
 }
-export default dataProvider;
\ No newline at end of file
+export default dataProvider;
+
+export const liveProvider = (client: SanityClient): LiveProvider => {
+  return {
+    subscribe({ callback, params, channel }) {
+      const subscription = client.listen(`*[_type == '${params?.resource}']`).subscribe((event) => {
+        if (event?.identity === '<system>') {
+          return
+        }
+        if (event?.documentId.includes('drafts.')) {
+          return
+        }
+
+        // Sanity sends WS events before the document is actually updated
```

Comment:
Hello @hirenf14, This was a bit of a hacky solution but unfortunately, I couldn't find another way. Do you have any idea how to solve this in a better way?

Reply:
Was this due to CDN usage? Because I think callback has to be called after its updated on their end.

1 option would be,
`useCdn: true, // set to `false` to bypass the edge cache`

By setting this to false, read operation won't use cached options. If that's the problem.
Otherwise, I think there should be some other way to catch it, but this method is not something we should deliver for performance.

Reply:
I tested with [`useCdn:false`](https://github.com/refinedev/refine/pull/5145/files#diff-f7f38799d844d0479dda90f0d558ac25e6b0c3321482e71130206a4499b0873eR28). 
Unfortunately, I couldn't find any other setting for this.

Reply:
Sorry, but I think this isn't ready to be rolled out. I hope you understand me blocking it.

Reply:
I'll be looking into it, in depth meanwhile, so that we can make it move further.

Reply:
Thanks. 
Btw, We created [an example](https://github.com/refinedev/refine/pull/5145) based on your data provider. Thanks for improving and refining 🚀